GraphBLAS/cmake_modules/GraphBLAS_JIT_paths.cmake

```diff
@@ -43,8 +43,20 @@ if ( SUITESPARSE_HAS_CUDA AND GRAPHBLAS_USE_CUDA )
 #   set ( GRAPHBLAS_COMPACT on )    # FIXME
 endif ( )
 
+include ( CheckSymbolExists )
+check_symbol_exists ( system "stdlib.h" HAVE_C_SYSTEM )
+
 option ( GRAPHBLAS_COMPACT "ON: do not compile FactoryKernels.  OFF (default): compile FactoryKernels" OFF )
-option ( GRAPHBLAS_USE_JIT "ON (default): use the CPU JIT.  OFF: do not use the CPU JIT" ON )
+option ( GRAPHBLAS_USE_JIT "ON (default): use the CPU JIT.  OFF: do not use the CPU JIT" ${HAVE_C_SYSTEM} )
+
+if ( GRAPHBLAS_USE_JIT AND NOT HAVE_C_SYSTEM )
```

Comment:
It seems odd that at line 50 the **GRAPH_USE_JIT** option is set according to if the above symbol check for system passes then to verify **HAVE_C_SYSTEM**.

To me it seems that **if ( GRAPHBLAS_USE_JIT )** does the job. Otherwise keep this line 52 as is but remove the line 50.

I'll state that the check_symbol_exists from the stdlib.h is a nice technique. I've forgotten about this approach in CMake. I like it since it adapts to potentially other platforms & scenarios. Good job.

Reply:
The GRAPHBLAS_USE_JIT is a user-selectable option. It tells use "the end user would like to use the JIT in GraphBLAS".  The default setting is ON if the system() call is available (the normal case).

If the setting is ON but the JIT fails, then we either (a) disable the JIT, or (b) report an error.  The latter occurs if SUITESPARSE_USE_STRICT is ON.  That's what line 52 is checking (and line 53).

Normally for these kind of options we have GRAPHBLAS_USE_something (a request from the end user) and GRAPHBLAS_HAS_something (if the option has been requested and successfully fulfilled).  So something can be requested (with _USE_ being ON) but we can't do it (_HAS_ is OFF).

Reply:
The GRAPHBLAS_USE_JIT is just a single variable but that looks fine to me.

Reply:
The last argument of `option` in line 50 is the *default* value for `GRAPHBLAS_USE_JIT`. The user might have set a different value, e.g., by adding `-DGRAPHBLAS_USE_JIT=ON` in the configure step. That would override the default value.

Line 52 is there to check if the build system should react on such a "known bad" configuration.